xmtp_mls/README.md

```diff
@@ -122,41 +116,58 @@ The [following diagram](https://app.excalidraw.com/s/4nwb0c8ork7/6pPH1kQDoj3) il
 
 - Key updates
 - Processing incoming welcome messages
+- Tracking group membership at the account/user level
+- Permissioning for adding/removing accounts/users
+- Mechanism for syncing installations under each account/user
 
 ### Add members to a group
 
 Simplified high level flow for adding members to a group:
 
-1. Consume Key Packages for all new members
 1. Create a `group_intent` for adding the members
-1. Sync the state of the group with the network
-1. Convert the intent into concrete commit and welcome messages for the current epoch
+1. Consume Key Packages for all new members
+1. Convert the intent (and any pending proposals) into concrete commit and welcome messages for the current epoch
+   1. Write the welcome messages to the `post_commit_data` field for later
 1. Publish commit message
 1. Sync the state of the group with the network
 1. If no conflicts: Publish welcome messages to new members.
-   If conflicts: Go back to step 4 and try again
+   If conflicts: Go back to step 2 and try again (reset the intent's state to `TO_SEND` and clear the `data` and `post_commit_data` fields)
 
 ### Remove members from a group
 
 Simplified high level flow for removing members from a group:
 
 1. Create a `group_intent` for removing the members
-1. Sync the state of the group with the network
-1. Convert the intent into concrete commit for the current epoch
+1. Convert the intent (and any pending proposals) into concrete commit for the current epoch
+   1. If there are welcome messages, write them to the `post_commit_data` field for later
 1. Publish commit to the network
 1. Sync the state of the group with the network
-1. If no conflicts: Done.
-   If conflicts: Go back to step 3 and try again
+1. If no conflicts: Done. If there are welcome messages in the `post_commit_data` field, publish them.
+   If conflicts: Go back to step 2 and try again (reset the intent's state to `TO_SEND` and clear the `data` and `post_commit_data` fields)
 
 ### Send a message
 
 Simplified high level flow for sending a group message:
 
 1. Create a `group_intent` for sending the message
+1. Commit any pending proposals
```

Comment:
One thing that gets a little weird about this flow is that committing any pending proposals will result in a new message to the network containing the commit in addition to the one containing the message. All other intents map to a single unprocessed message on the network. That breaks the model a bit (what `payload_hash` do we store on the intent?). Maybe the thing to do here is actually to spawn a new intent committing the change? Open to other ideas.